src/test/java/com/es/phoneshop/dao/impl/ArrayListOrderDaoTest.java

```diff
@@ -0,0 +1,114 @@
+package com.es.phoneshop.dao.impl;
+
+import com.es.phoneshop.exception.ArgumentIsNullException;
+import com.es.phoneshop.exception.NotFoundException;
+import com.es.phoneshop.exception.OrderNotFoundException;
+import com.es.phoneshop.model.order.Order;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.UUID;
+
+import static org.junit.Assert.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ArrayListOrderDaoTest {
+    private ArrayListOrderDao orderDao;
+    private Long id = 1L;
+    Order order1;
+    Order order2;
+    String secureId1;
+    String secureId2;
+
+    @Before
+    public void setup() throws NotFoundException {
+        orderDao = ArrayListOrderDao.getInstance();
+        secureId1 = UUID.randomUUID().toString();
+        secureId2 = UUID.randomUUID().toString();
+        order1 = new Order();
+        order1.setSecureId(secureId1);
+        order2 = new Order();
+        order2.setSecureId(secureId2);
+        orderDao.save(order1);
+        orderDao.save(order2);
+    }
+
+    @Test
+    public void testGetOrder() throws NotFoundException {
+        assertEquals(order1, orderDao.getItem(id));
+    }
+
+    @Test(expected = NotFoundException.class)
+    public void testGetOrderNotFound() throws NotFoundException {
+        Long id = 0L;
+        orderDao.getItem(id);
```

Comment:
Here and in similar tests you could also validate what is the class inside NotFoundException.